src/main/java/org/commonwl/view/git/GitDetails.java

```diff
@@ -242,4 +252,44 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(repoUrl, branch, path, packedId);
     }
+
+    /**
+     * Retrieves license details from the repo, if present.
+     * @param workTree the path to the locally cloned repo
+     * @return The license URI
+     */
+    public String getLicense(Path workTree) {
+        try {
+            String[] command = {"licensee", "detect", "--json", workTree.toString()};
+            if (logger.isDebugEnabled()) {
+                logger.debug("Calling " + String.join(" ", command));
+            }
+            Process process = Runtime.getRuntime().exec(command, null);
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode jsonLicenses = mapper.readTree(process.getInputStream());
+            if (logger.isDebugEnabled()) {
+                logger.debug("Licensee retrieved the following licenses:\n" + jsonLicenses.toPrettyString());
+            }
+            int size = jsonLicenses.withArray("licenses").size();
+            if (size > 0) {
+                String licenseCandidate = jsonLicenses.withArray("matched_files")
+                        .get(0).get("filename").asText();
+                String licenseLink = getRawUrl(null, licenseCandidate);
+                if (logger.isWarnEnabled() && size > 1) {
+                    logger.warn("There are " + size + " identified license files in the " + repoUrl + " repository. "
+                            + "Taking the first one: " + licenseLink);
+                }
+                String key = jsonLicenses.withArray("licenses").get(0).get("key").asText();
+                if (!"other".equals(key)) {
+                    return SPDX_PREFIX + jsonLicenses.withArray("licenses").get(0).get("spdx_id").asText();
+                } else {
+                    return licenseLink;
+                }
+            } else {
+                return null;
+            }
+        } catch(IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Mention some more context, for the logs please:
```suggestion
            throw new RuntimeException("While attempting to detect license for " + workTree.toString + ": " + e.getMessage(), e);
```

Reply:
Added a proper `GitLicenseException` class